spring-kafka-avro/src/main/java/com/example/kafka/ThingAvroSerde.java

```diff
@@ -0,0 +1,125 @@
+package com.example.kafka;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.Decoder;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificRecord;
+import org.apache.kafka.common.errors.SerializationException;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serializer;
+
+public class ThingAvroSerde implements Serializer<SpecificRecord>, Deserializer<Object> {
+
+	@Override
+	public Object deserialize(String topic, byte[] data) {
+		return null;
+	}
+
+	@Override
+	public void configure(Map<String, ?> configs, boolean isKey) {
+		Serializer.super.configure(configs, isKey);
+	}
+
+	@Override
+	public Object deserialize(String topic, Headers headers, byte[] data) {
+		try {
+			Object result = null;
+
+			if (data != null) {
+				DatumReader<GenericRecord> datumReader;
+				switch (new String(headers.lastHeader("thing").value())) {
+					case "Thing1":
+						datumReader = new SpecificDatumReader<>(
+								Thing1.class.getDeclaredConstructor().newInstance().getSchema());
```

Comment:
I wonder why there is reflection used here. Can't this class be `new`ed?

Reply:
I wanted to emulate how the Confluent `KafkaAvroDeserializer` works (which I can't use because it uses an older jackarta.ws.rs jar with the `javax` package namespace). However, since the deserializer is not a bean, the type won't be detected anyway. Will change.